example/client.rb

```diff
@@ -68,7 +68,9 @@
 
 stream.on(:close) do
   log.info 'stream closed'
-  sock.close
+
+  # keep connection open to handle push messages
+  # sock.close
```

Comment:
No, this is not correct. The parent stream must be open for PUSH_PROMISE to be valid; the server can't arbitrarily push against closed and/or non-existent streams.